src/Buttons.js

```diff
@@ -3,23 +3,35 @@ class Buttons {
     this._buttons = [];
   }
 
-  add({text, data, url, event}) {
-    if (!data && !url && !event) {
+  add({text, data, url, phone, event, share, account_linking}) {
+    if (!data && !url && !event && !phone && !share) {
       throw Error('Must provide a url or data i.e. {data: null} or {url: \'https://facebook.com\'}');
     }
 
-    this._buttons.push({text: text || 'Button', event, data, url});
+    this._buttons.push({text: text || 'Button', event, data, phone, share, url, account_linking});
     return this;
   }
 
   toJSON() {
     const buttons = [];
     for (const button of this._buttons) {
-      if (button.url) {
+      if (button.account_linking) {
+        if (!button.account_linking.linking) {
+          buttons.push({type: 'account_unlink'});
+        } else if (button.url) {
+          buttons.push({type: 'account_link', url: button.url});
+        } else {
+          console.log('[fb-msger-bot] Missing url for account linking');
```

Comment:
Throw an error instead? Throwing error pattern is preferred when it has to be used.